src/github/client/github-client.ts

```diff
@@ -158,4 +160,16 @@ export default class GitHubClient {
 		return response?.data?.data?.viewer?.repositories?.totalCount as number;
 	}
 
+
+	public async getBranchesPage(owner: string, repoName: string, perPage?: number, cursor?: string) : Promise<Repository> {
+		const response = await this.graphql(ViewerRepositoryCountQuery,
+			{
+				owner: owner,
+				repo: repoName,
+				per_page: perPage,
+				cursor
+			});
+		return response?.data?.data.repository as Repository;
```

Comment:
Wouldn't it be better to make it `this.graphql<{repository:Repository}>` then just return the data?  Could make it inline too:

```
return (await this.graphql<{repository:Repository}>(ViewerRepositoryCountQuery,
			{
				owner: owner,
				repo: repoName,
				per_page: perPage,
				cursor
			}))?.data?.data.repository;
```

I'd probably put the return type of graphql next to the query in the other file so it's easier to match up.

Reply:
ex: `await this.graphql<ViewerRepositoryCountQueryType>(ViewerRepositoryCountQuery)`

Reply:
So you want me to make graphql function gneric, right?

In the previous PR it was, but I was getting some typecript errors and reverted back to what it was. I can look into restoring it